src/vs/workbench/browser/parts/notifications/notificationsTelemetry.ts

```diff
@@ -0,0 +1,55 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Disposable } from 'vs/base/common/lifecycle';
+import { INotificationService } from 'vs/platform/notification/common/notification';
+import { ITelemetryService } from 'vs/platform/telemetry/common/telemetry';
+import { IWorkbenchContribution, IWorkbenchContributionsRegistry, Extensions } from 'vs/workbench/common/contributions';
+import { Registry } from 'vs/platform/registry/common/platform';
+import { LifecyclePhase } from 'vs/workbench/services/lifecycle/common/lifecycle';
+import { hash } from 'vs/base/common/hash';
+
+interface NotificationMetrics {
+	id: number;
+	source?: string;
+}
+
+type NotificationMetricsClassification = {
+	id: { classification: 'SystemMetaData', purpose: 'FeatureInsight' };
+	source?: { classification: 'SystemMetaData', purpose: 'FeatureInsight' }
+};
+
+class NotificationsTelemetryContribution extends Disposable implements IWorkbenchContribution {
+
+	constructor(
+		@ITelemetryService telemetryService: ITelemetryService,
+		@INotificationService notificationService: INotificationService
+	) {
+		super();
+		this._register(notificationService.onDidAddNotification(notification => {
+			if (!notification.silent) {
+				telemetryService.publicLog2<NotificationMetrics, NotificationMetricsClassification>('notification:show', {
+					id: hash(notification.message.toString()),
+					source: notification.source
+				});
+			}
+		}));
+
+		this._register(notificationService.onDidRemoveNotification(notification => {
+			if (!notification.silent) {
```

Comment:
When I remove a toast though, the `silent` property shouldn't matter? Or do we want to keep showing and removing in sync?

Reply:
Yeah I though having showing and removing in sync makes sense. And that silent notifications are not "spammy" and we could ignore them for data collection.
Let's see what @devinvalenciano thinks about `silent` notifications

Reply:
Sorry missing some context here - what is a silent notification?  What does that look like for the end-user?

From a telemetry perspective, is it possible to have an additional silent -> (true, false) flag? that way we retain the signal but can filter to silent/non-silent notifications as needed?

Reply:
@devinvalenciano silent notifications are only shown in the status bar and do not pop out. They only pop out if the user clicks on the notification center.
Yes we can add a `silent` flag to the telemetry event. Just confirm here that's what you would prefer and I will add that flag and report silent notifiation

Example Initializing TS/JS lang server: 

![Screenshot 2021-03-16 at 19 14 08](https://user-images.githubusercontent.com/1926584/111359444-cd6d4a80-868b-11eb-9098-f3d5909f9e9c.png)

Reply:
Thanks for the example @isidorn -

I think retaining this information would be helpful, but entirely separate from analyzing toast notifications.  For example, we may want to understand the effectiveness/action-rate/time-to-action on these silent notifications down the line.

Would this be how a silent notification works in the new telemetry flow:
- Notification appears in status bar, VS Code fires notification:shown with silent==true.
- User clicks on silent notification, VS Code fires notification:actionExecuted with silent == true.  What would `actionId` be here?
- Notification disappears naturally, VS Code fires notification:close

Let me know if I'm thinking about this incorrectly.

Reply:
@devinvalenciano yes you are thinking about it properly.
So I will add a `silent` field to the telemetry data.

actionExecuted will not have the `silent==true`, you can get that from the show event with which you can corelate the actionExecuted one. 
The actionId, or actionLabel would only be there if a silent notification has a button. For example long running explorer file operations have a Cancel button.

Reply:
I see, this makes perfect sense to me!  Thanks for walking me through this concept.